command/ui_input.go

```diff
@@ -119,10 +121,21 @@ func (i *UIInput) Input(opts *terraform.InputOpts) (string, error) {
 	// interrupt this if we are interrupted (SIGINT)
 	result := make(chan string, 1)
 	go func() {
-		buf := bufio.NewReader(r)
-		line, err := buf.ReadString('\n')
-		if err != nil {
-			log.Printf("[ERR] UIInput scan err: %s", err)
+		var line string
+		var err error
+
+		if opts.Secret == "true" {
+			secretBytes, err := terminal.ReadPassword(0)
```

Comment:
It seems like this would not respect the situation where variable `r` is set to something other than `os.Stdin`. Do I understand correctly here?

Overridden readers and writers can be useful in tests, so it would be ideal to still support this being overridden here; I'm not too familiar with the `terminal` library here, but it looks like the `0` here is a file descriptor, in which case maybe there's a solution here:

* Check if `r` is an instance of `*os.File`
* If it is, call the `Fd` method to get the file descriptor and pass it here.
* If it isn't, use the `bufio.NewReader` technique as if this is _not_ a sensitive value, just so there's still a way to pass in data from a non-file reader if we need to in tests.

Does this seem feasible?